core/domain/exp_domain.py

```diff
@@ -2101,6 +2110,71 @@ def _convert_states_v25_dict_to_v26_dict(cls, states_dict):
 
         return states_dict
 
+    @classmethod
+    def _convert_states_v26_dict_to_v27_dict(cls, states_dict):
+        """Converts from version 26 to 27. Version 27 adds content_translations
+        dict to the state, which will allow translators to add translation
+        script for the state contents.
+
+        NOTE: This migration will also filter out the content_id from
+        content_ids_to_audio_translations such that the state passes the new
+        validation check safely.
+
+        Earlier state validation use to check that the set of all content ids
+        present within the state is sub-set of the
+        content_ids_to_audio_translations keys, but the new validation will
+        check whether both are equal.
```

Comment:
Can you manually check that there is no user-facing breakage for existing explorations and existing editing-sessions-in-progress when you switch from the develop branch to this branch? Use an exploration (created on top of develop branch) where the set of content IDs is actually a proper subset (as described here).

Reply:
Working on it!